### ISSUE TITLE
Can't open a new trade for EOS/USDT: stake amount is too small

### ISSUE THREAD

**Issue by leonardvdj** [freqtrade/freqtrade#4398]
I recently switched to USDT because I wanted to try that, as I didn't want to rely on the volatility of BTC as a base currency, however I'm running into an issue.
I have 115 USDT in my binance wallet, max open trades is 6, and my stake amount is 15, however when my strategy produces a buy signal I run into this:
```freqtrade        | 2021-02-19 02:14:40,809 - freqtrade.wallets - INFO - Wallets synced.
freqtrade        | 2021-02-19 02:14:40,812 - freqtrade.freqtradebot - INFO - Buy signal found: about create a new trade with stake_amount: 15 ...
freqtrade        | 2021-02-19 02:14:40,813 - freqtrade.freqtradebot - INFO - Executing Buy for EOS/USDT
freqtrade        | 2021-02-19 02:14:40,813 - freqtrade.freqtradebot - INFO - Getting price from order book Ask side.
freqtrade        | 2021-02-19 02:14:41,290 - freqtrade.freqtradebot - INFO - ...top 1 order book buy rate 4.72780000
freqtrade        | 2021-02-19 02:14:41,290 - freqtrade.freqtradebot - WARNING - Can't open a new trade for EOS/USDT: stake amount is too small (15 < 15.384615384615387)
```

Bot version 2021.1, running via docker on Ubuntu 20.04.2 LTS, edge is disabled.

I can't exactly see why it's doing this, as I went and tested on binance and could place buy orders for EOS for even just 10 USDT, so its not a limitation of binance.
Any reason why this could be?

**Comment by xmatthias** [comment]
Freqtrade includes stoploss into the minimum order calculation - basically making sure it's still able to place a stoploss order below your entry price.

So while buying at 15$ will still work (the minimum is usually around 10$ - but there's a minimum amount for the coin as well, so the price plays a role, too) - if the price falls by "stoploss + some margin" - it would fall below this minimum, disallowing the bot to sell the coin.

in theory, not beeing able to sell becomes then a loss of 100% - or it could be if the coin goes away or falls VERY low.

**Comment by leonardvdj** [comment]
Yeah that's definitely it then, got a very moderate stoploss. I guess I'll play around with it some more, thank you! :)

**Comment by leonardvdj** [comment]
I'm reopening this because something occurred to me yesterday. The minimum, for USDT at least, is $10. As observed above it put the limit at 15.384, so lets for simplicity sake say 50% above the minimum, which corresponds with my 50% stoploss.
Shouldn't this calculation actually be minimum / stoploss, so 10 / 0.5 to give $20, not minimum * (1 + stoploss) equaling $15? If I hit stoploss on $15, I won't be able to sell, but if I hit stoploss on $20 I will just barely be able to.
Maybe there's something I'm missing though, and I haven't tested if the buy and sell minimums are different but I'd assume they both lie at $10.

**Comment by xmatthias** [comment]
well the worst case is 50% (0.5) - but in almost all other cases, it'll be below the 20$.

there are multiple "minimum" stake amounts - one for cost, and one per amount  (the amount one will depend on the price, but is (almost always) below the cost limit.
So let's assume a cost-limit of 10$ (as documented on the binance page).

by default, a small % (5%) is reserved as "amount_reserve_percent".
If you're trading with values very close to the limit, you'll be happy to have this buffer - assuming a stoploss order will also need a bit of room below to actually fill (for stoploss on exchange) - or to be placed (without on exchange).

so the real calculation will be:

`highest of limits / (1.0 - (amount_reserve_percent) - stoploss (which is negative))`

`10 / (1.0 - 0.05 - 0.5) = 22$`

we do however limit the divisor to 0.5 - so it'll never be more than 20$.

`10 / max(1.0 - 0.05 - 0.5, 0.5) = 20$`.

Now obviously, if you use a bigger stoploss than 0.5, this calculation won't work anymore - but we can't reserve "unlimited" space below either (assume you don't want to use stoploss at all and set it to 200).

**Comment by leonardvdj** [comment]
What I'm failing to understand though, is how it landed at 15.384, and not $22 or $20 as you described above.

**Comment by xmatthias** [comment]
well limits change over time eventually - and it's ultimately based on a response from the API - so it's fully possible that the API didn't report 10$, but something slightly lower for this particular pair.

You can look at the code yourself - you'll see it'll be exactly what i described above.

https://github.com/freqtrade/freqtrade/blob/4b689f5b887e2b8d138368e2aac796d1874156b2/freqtrade/exchange/exchange.py#L507-L540
